util/errors/errors.go

```diff
@@ -15,7 +15,7 @@ func IsTransientErr(err error) bool {
 		return false
 	}
 	err = argoerrs.Cause(err)
-	return isExceededQuotaErr(err) || apierr.IsTooManyRequests(err) || isResourceQuotaConflictErr(err) || isTransientNetworkErr(err)
+	return isExceededQuotaErr(err) || apierr.IsTooManyRequests(err) || isResourceQuotaConflictErr(err) || isTransientNetworkErr(err) || apierr.IsServerTimeout(err) || apierr.IsServiceUnavailable(err)
```

Comment:
needs test obvs

Reply:
@jessesuen I think I recall we excluded timeout as transient error - but I feel it does fit the definition - can you remember why?

Reply:
@dtaniwaki I spoke to Jesse. We need to be careful here. We call this func in a poll or back-off loop. A timeout retried multiple times would end up being VERY long in those cases. However, whenever we do that, the error would result in a failed workflow - which should have been retried. Therefore I think this change is correct.